### PR TITLE
fix: #1757 fix set MAX_CREATION_LIMIT as zero.

### LINKED ISSUE #1757: MAX_CREATION_LIMIT
- Gitea version (or commit ref):25d6e2a
- Git version:2.11.2
- Operating system:ubuntu/docker
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description
I am using the latest docker image for hosting. Have used the Postgresql linking method to setup my docker gitea installation.
After a normal setup, I login with my newly created admin account.
Then I setup SMTP source auth and change my own auth source to the newly created source.
Now I want to limit every new user that logs in to not be able to create any repositories.
Updating each and every user's setting is not feasible.
As per [documentation](https://docs.gitea.io/en-us/config-cheat-sheet/), setting MAX_CREATION_LIMIT to zero should do the trick.

So I went back to the docker image `docker exec -it gitea`, performed opened the `/data/gitea/conf/app.ini` and set `MAX_CREATION_LIMIT=0` in the repositories section just below the path variable.

Restarting with `docker restart gitea` though, has no affect on newly logged in users (via the SMTP source)

### PR DESCRIPTION
**Pull request by appleboy**
fix #1757 

as title.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Comment by sapk** [comment]
LGTM

File: modules/setting/setting.go [hunk 1]
```diff
@@ -871,6 +871,7 @@ please consider changing to GITEA_CUSTOM`)
 	// Determine and create root git repository path.
 	sec = Cfg.Section("repository")
 	Repository.DisableHTTPGit = sec.Key("DISABLE_HTTP_GIT").MustBool()
+	Repository.MaxCreationLimit = sec.Key("MAX_CREATION_LIMIT").MustInt()
```
Comment (lafriks) [inline modules/setting/setting.go]: Shouldn't default be -1 - unlimited?

Comment (bkcsoft) [reply modules/setting/setting.go]: ☝️

Comment (appleboy) [reply modules/setting/setting.go]: @lafriks @bkcsoft Done.

**Comment by lafriks** [comment]
LGTM

**Pull request merged**